packages/dbos-cloud/applications/deploy-app-code.ts

```diff
@@ -218,3 +239,85 @@ function readInterpolatedConfig(configFilePath: string, logger: CLILogger): stri
     return "";
   });
 }
+
+async function isAppRegistered(logger: Logger, host: string, appName: string, userCredentials: DBOSCloudCredentials): Promise<boolean> {
+  let appRegistered = true;
+  const bearerToken = "Bearer " + userCredentials.token;
+  try {
+    await axios.get(`https://${host}/v1alpha1/${userCredentials.organization}/applications/${appName}`, {
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: bearerToken,
+      },
+    });
+  } catch (e) {
+    const errorLabel = `Failed to retrieve info for application ${appName}`;
+    const axiosError = e as AxiosError;
+    if (isCloudAPIErrorResponse(axiosError.response?.data)) {
+      const resp: CloudAPIErrorResponse = axiosError.response?.data;
+      if (resp.message.includes(`application ${appName} not found`)) {
+        appRegistered = false;
+      } else {
+        handleAPIErrors(errorLabel, axiosError);
+      }
+    } else {
+      logger.error(`${errorLabel}: ${(e as Error).message}`);
+    }
+  }
+  return appRegistered;
+}
+
+async function chooseAppDBServer(logger: Logger, host: string, userCredentials: DBOSCloudCredentials): Promise<string> {
+  // List existing database instances.
+  let userDBs: UserDBInstance[] = [];
+  const bearerToken = "Bearer " + userCredentials.token;
+  try {
+    const res = await axios.get(`https://${host}/v1alpha1/${userCredentials.organization}/databases`, {
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: bearerToken,
+      },
+    });
+    userDBs = res.data as UserDBInstance[];
+  } catch (e) {
+    const errorLabel = `Failed to list databases`;
+    const axiosError = e as AxiosError;
+    if (isCloudAPIErrorResponse(axiosError.response?.data)) {
+      handleAPIErrors(errorLabel, axiosError);
+    } else {
+      logger.error(`${errorLabel}: ${(e as Error).message}`);
+    }
+    return "";
+  }
+
+  let userDBName = "";
+  if (userDBs.length === 0) {
+    // If not, prompt the user to provision one.
+    logger.info("No database found, provisioning a database server...");
```

Comment:
Maybe say database instance?  It's what we say everywhere else.

Reply:
Yeah it's a confusing thing in Postgres. I picked server because the PG docs uses this term: https://www.postgresql.org/docs/current/server-start.html

> Before anyone can access the database, you must start the database server.

I'm thinking to update the term everywhere else. "Instance" is vague.

Reply:
I'll change it to database instance (server).